catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/activity/rtl/MultilingualTest.java

```diff
@@ -0,0 +1,138 @@
+ */
+
+package org.catrobat.catroid.uiespresso.activity.rtl;
+
+import android.support.test.runner.AndroidJUnit4;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.ui.Multilingual;
+import org.catrobat.catroid.uiespresso.util.BaseActivityInstrumentationRule;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.Locale;
+
+import static android.support.test.espresso.Espresso.onData;
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.assertThat;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static junit.framework.Assert.assertTrue;
+
+import static org.catrobat.catroid.uiespresso.util.UiTestUtils.getResourcesString;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.hasToString;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.startsWith;
+
+@RunWith(AndroidJUnit4.class)
+public class MultilingualTest {
+	private static final Locale ARABICLOCALE = new Locale("ar");
+	private static final Locale URDULOCALE = new Locale("ur");
+	private static final Locale FARSILOCALE = new Locale("fa");
+	private static final Locale DEUTSCHLOCALE = Locale.GERMAN;
+	@Rule
+	public BaseActivityInstrumentationRule<Multilingual> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(Multilingual.class);
+
+	@Before
+	public void setUp() throws Exception {
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	@Test
+	public void testChangeLanguageToDeutsch() {
+		onData(hasToString(startsWith(DEUTSCHLOCALE.getDisplayName(DEUTSCHLOCALE)))).perform(click());
+		assertThat("current Locale is not Deutsch", Locale.getDefault().getDisplayLanguage(), is(DEUTSCHLOCALE.getDisplayLanguage()));
+		String buttonProgramsName = getResourcesString(R.string.main_menu_programs);
+		String buttonContinueName = getResourcesString(R.string.main_menu_continue);
+		String buttonNewName = getResourcesString(R.string.main_menu_new);
+		onView(withId(R.id.main_menu_button_continue))
+				.check(matches(withText(containsString(buttonContinueName))));
+		onView(withId(R.id.main_menu_button_new))
+				.check(matches(withText(buttonNewName)));
+		onView(withId(R.id.main_menu_button_programs))
+				.check(matches(withText(buttonProgramsName)));
+	}
+
+	@Test
+	public void testChangeLanguageToArabic() {
+		onData(hasToString(startsWith(ARABICLOCALE.getDisplayName(ARABICLOCALE)))).perform(click());
+		assertThat("current Locale is not Arabic", Locale.getDefault().getDisplayLanguage(), is(ARABICLOCALE.getDisplayLanguage()));
+		assertTrue("current Text direction is not RTL", isRTL(Locale.getDefault()));
```

Comment:
if the right assertion is chosen, assertion messages do not add any value.
here for example theres no point in the message, since it should be clear on error output (if it ever fails), that isRTL was returning false..